core/src/main/scala/cats/data/NonEmptyMap.scala

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2018 Luka Jacobowitz
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cats
+package data
+
+import cats.instances.sortedMap._
+import cats.kernel._
+import cats.{Always, Apply, Eval, Foldable, Functor, Later, NonEmptyTraverse, Now, SemigroupK, Show}
+
+import scala.collection.immutable._
+
+final class NonEmptyMap[K, A] private(val value: SortedMap[K, A]) extends AnyVal {
```

Comment:
does it have to be a `SortedMap`?

Reply:
What would you want it to be?

Reply:
just `immutable.Map[K,A]`?
or maybe a better name for the class then `NonEmptySortedMap`?